dsa/src/components.rs

```diff
@@ -81,7 +86,9 @@ impl<'a> DecodeValue<'a> for Components {
         let q = BigUint::from_bytes_be(q.as_bytes());
         let g = BigUint::from_bytes_be(g.as_bytes());
 
-        Ok(Self::from_components(p, q, g))
+        // TODO: Using this error doesn't seem right
+        Self::from_components(p, q, g)
+            .map_err(|_| der::Error::new(ErrorKind::Value { tag: Tag::Integer }, reader.position()))
```

Comment:
This can more succinctly be written as:

```suggestion
            .map_err(|_| reader.error(Tag::Integer.value_error()))
```